suzieq/engines/pandas/engineobj.py

```diff
@@ -75,6 +77,21 @@ def _handle_user_query_str(self, df: pd.DataFrame,
 
         return df
 
+    def _is_in_subnet(self, addr: PandasObject, net: str) -> PandasObject:
+        """Check if the IP addresses in a Pandas dataframe
+        belongs to the given subnet
+
+        Args:
+            addr (PandasObject): the collection of ip addresses to check
+            net: (str): network id of the subnet
+
+        Returns:
+            PandasObject: A collection of bool reporting the result
+        """
+        return addr.apply(lambda a: (
+            False if not a else ip_address(a.split("/")[0]) in ip_network(net))
```

Comment:
To improve the efficiency of not computing the network each time, can you pull out the ```ip_network(net)``` to be outside this function and pass it the output of ```ip_network(net)``` ?